File: Chapter-04-Strings-AssociativeArrays/Remove-Shorter-Strings/Remove-Shorter-Strings.js
// Given a string array and a value, remove any strings shorter than value from the array

function removeShortString(str, len){
  for (let i = 0; i < str.length; i++){
    if (len > str[i].length){
      str.splice(i, 1)
      i--
    }
  }
}

var myStr = ['this', 'is', 'a', 'string', 'array'],
    len   = 3

console.log(`The string array is '${myStr}'`)
console.log(`The given length is '${len}'`)
removeShortString(myStr, len)
console.log(`The string array after removing shorter strings is '${myStr}'`)

File: Chapter-04-Strings-AssociativeArrays/Remove-Shorter-Strings/Remove-Shorter-Strings.py
# Given a string array and a value, remove any strings shorter than value from the array

def removeShortString(stringArr, length):
    i = 0
    while (i < len(stringArr)):
        if length > len(stringArr[i]):
            stringArr.remove(stringArr[i])
        else:
            i += 1

myStrArr = ['this', 'is', 'a', 'string', 'array']
length = 3

print("The string array is {}").format(myStrArr)
print("The given length is {}").format(length)
removeShortString(myStrArr, length)
print("The string array after removing the shorter strings is {}").format(myStrArr)
